libs/community/langchain_community/vectorstores/vdms.py

```diff
@@ -0,0 +1,1630 @@
+            Lower score represents more similarity.
+        """
+        if self.embedding_function is None:
+            raise ValueError("Must provide embedding function")
+        else:
+            query_embedding: List[float] = self._embed_query(query)
+            results = self.query_collection_embeddings(
+                query_embeddings=[query_embedding],
+                n_results=k,
+                fetch_k=fetch_k,
+                filter=filter,
+                **kwargs,
+            )
+
+        return _results_to_docs_and_scores(results)
+
+    def similarity_search_with_score_by_vector(
+        self,
+        embedding: List[float],
+        k: int = DEFAULT_K,
+        fetch_k: int = DEFAULT_FETCH_K,
+        filter: Optional[Dict[str, List]] = None,
+        **kwargs: Any,
+    ) -> List[Tuple[Document, float]]:
+        """
+        Return docs most similar to embedding vector and similarity score.
+
+        Args:
+            embedding (List[float]): Embedding to look up documents similar to.
+            k (int): Number of Documents to return. Defaults to 3.
+            fetch_k (int): Number of candidates to fetch for knn (>= k).
+            filter (Optional[Dict[str, str]]): Filter by metadata. Defaults to None.
+
+        Returns:
+            List[Tuple[Document, float]]: List of documents most similar to
+            the query text and cosine distance in float for each.
+            Lower score represents more similarity.
+        """
+        kwargs["normalize_distance"] = True
+
+        results = self.query_collection_embeddings(
+            query_embeddings=[embedding],
+            n_results=k,
+            fetch_k=fetch_k,
+            filter=filter,
+            **kwargs,
+        )
+        return _results_to_docs_and_scores(results)
+
+    def update_document(
+        self, collection_name: str, document_id: str, document: Document
+    ) -> None:
+        """Update a document in the collection.
+
+        Args:
+            document_id (str): ID of the document to update.
+            document (Document): Document to update.
+        """
+        return self.update_documents(collection_name, [document_id], [document])
+
+    def update_documents(
+        self, collection_name: str, ids: List[str], documents: List[Document]
+    ) -> None:
+        """Update a document in the collection.
+
+        Args:
+            ids (List[str]): List of ids of the document to update.
+            documents (List[Document]): List of documents to update.
+        """
+        text = [document.page_content for document in documents]
+        metadata = [
+            validate_vdms_properties(document.metadata) for document in documents
+        ]
+        embeddings = self._embed_documents(text)
+
+        self.__update(
+            collection_name,
+            ids,
+            metadatas=metadata,
+            embeddings=embeddings,
+            documents=text,
+        )
+
+
+"""
```

Comment:
nit: `"""` are typically not used for doc-strings except for module level, class level or function/method level docs.

I'd swap to the more standard `#` for this kind of documentation